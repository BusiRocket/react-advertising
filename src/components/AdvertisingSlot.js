import React from 'react';
import connectToAdServer from './utils/connectToAdServer';
import PropTypes from 'prop-types';

export function AdvertisingSlotDiv({ id, style, className, children }) {
    return <div id={id} style={style} className={className} children={children} />;
}

AdvertisingSlotDiv.propTypes = {
    id: PropTypes.string.isRequired,
    style: PropTypes.object,
    className: PropTypes.string,
    children: PropTypes.node
};

export default connectToAdServer(AdvertisingSlotDiv);
