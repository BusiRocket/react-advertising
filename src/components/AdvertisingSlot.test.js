import React from 'react';
import { spy } from 'sinon';
import renderer from 'react-test-renderer';
import { AdvertisingSlotDiv } from './AdvertisingSlot';

const id = 'my-beautiful-id';
const className = 'large pink';
const styles = {
    color: 'hotpink',
    fontSize: '2em'
};

describe('The advertising slot component', () => {
    let mockConnectToAdServer;
    beforeEach(() => {
        jest.resetModules();
        mockConnectToAdServer = spy();
        jest.doMock('./utils/connectToAdServer', () => component => {
            mockConnectToAdServer(component);
            return component;
        });
        require('./AdvertisingSlot');
    });
    it('is connected to the ad server', () => void expect(mockConnectToAdServer).to.have.been.called);
});

describe('The div component wrapped by the advertising slot component', () => {
    describe('with just an ID', () => snapshotTest(<AdvertisingSlotDiv id={id} />));
    describe('with CSS classes', () => snapshotTest(<AdvertisingSlotDiv id={id} className={className} />));
    describe('with CSS styles', () => snapshotTest(<AdvertisingSlotDiv id={id} styles={styles} />));
    describe('with children', () =>
        snapshotTest(
            <AdvertisingSlotDiv id={id}>
                <h1>Hello!1!</h1>
            </AdvertisingSlotDiv>
        ));
});

function snapshotTest(component) {
    it('is rendered correctly', () => expect(renderer.create(component).toJSON()).toMatchSnapshot());
}
