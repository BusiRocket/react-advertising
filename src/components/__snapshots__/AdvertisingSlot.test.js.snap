// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The div component wrapped by the advertising slot component with CSS classes is rendered correctly 1`] = `
<div
  className="large pink"
  id="my-beautiful-id"
  style={undefined}
/>
`;

exports[`The div component wrapped by the advertising slot component with CSS styles is rendered correctly 1`] = `
<div
  className={undefined}
  id="my-beautiful-id"
  style={undefined}
/>
`;

exports[`The div component wrapped by the advertising slot component with children is rendered correctly 1`] = `
<div
  className={undefined}
  id="my-beautiful-id"
  style={undefined}
>
  <h1>
    Hello!1!
  </h1>
</div>
`;

exports[`The div component wrapped by the advertising slot component with just an ID is rendered correctly 1`] = `
<div
  className={undefined}
  id="my-beautiful-id"
  style={undefined}
/>
`;
